File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
import java.util.*;

class Solution {
    static String alph = "AEIOU";
    static int answer;
    static int count = -1;
    public int solution(String word) {
        dfs(0, word, "");
        return answer;
    }
    public void dfs(int depth, String word, String path){
        count++;
        // System.out.println(path + " " + count + " " + path.equals(word));
        if (path.equals(word)) {
            answer = count;
            return;
        }
        if (depth == 5) return;
        for (int i = 0; i < 5; i++){
            dfs(depth + 1, word, path + alph.charAt(i));
        }
    }
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
cnt = -1
answer = 0

def dfs(depth, word, cur):
    # print(cur)
    global cnt
    global answer
    cnt +=1
    if word == cur:
        answer = cnt
    if depth == 6:
        return
    dfs(depth + 1, word, cur + "A")
    dfs(depth + 1, word, cur + "E")
    dfs(depth + 1, word, cur + "I")
    dfs(depth + 1, word, cur + "O")
    dfs(depth + 1, word, cur + "U")
    

def solution(word):
    global answer
    dfs(1, word, "")
    print(answer)
    return answer